submissions/Dovahkiin1991/friends-app/app.js

```diff
@@ -0,0 +1,304 @@
+const usersList = [];
+let usersFilteredList = [];
+const nameInput = document.querySelector('.filter__block input');
+
+const callAPI = async () => {
+    const data = await fetch('https://randomuser.me/api/?results=30&nat=ua');
+    handleErrors(data);
+
+    const { results: users } = await data.json();
+
+    return users;
+};
+
+const handleErrors = (response) => {
+    if (!response.ok) {
+        throw Error(response.statusText);
+    }
+    return response;
+}
+
+const getUsers = async () => {
+    try {
+        const users = await callAPI();
+        usersList.push(...users);
+        renderUsers(getSidebarOptions(), usersList);
+    } catch (e) {
+        console.log(e);
+    }
+}
+
+const mainContainer = document.getElementById('main__content');
+const usersListContainer = document.createElement('div');
+usersListContainer.classList.add('users');
+
+const renderUsers = (filtersObject, usersArray) => {
+    let users = filterUsers(filtersObject, usersArray);
+    users = users.map((user) => createUser(user)).join("");
+    usersListContainer.innerHTML = users;
+    mainContainer.innerHTML = '';
+    mainContainer.appendChild(usersListContainer);
+    window.location.hash = '';
+}
+
+const createUser = (user) => {
+    let picture = user.picture.medium;
+    let name = user.name.first + ' ' + user.name.last;
+    let age = user.dob.age;
+    let location = user.location.country + ', ' + user.location.city;
+    let email = user.email;
+    let phone = user.phone;
+    let gender = user.gender;
+
+    return `
+        <div class="users__item">
+            <div class="user">
+                <div class="user-block">
+                    <div class="user-poster">
+                        <img src="${picture}" alt="${name}">
+                    </div>
+
+                    <div class="user-headings">
+                        <h5><strong>${name}</strong></h5>
+                        <h6>${age} years old</h6>
+                    </div>
+
+                    <div class="user-description">
+                        <p>${location}</p>
+                        <p>
+                            <a href="mailto:${email}">${email}</a>
+                        </p>
+                        <p>
+                            <a href="tel:${phone}">${phone}</a>
+                        </p>
+                        <p class="uppercase">${gender}</p>
+                    </div>
+
+                    <div class="user-button" data-id="${name.replace(/\s/g, '').toLowerCase()}">
+                        View Profile
+                    </div>
+                </div>
+            </div>
+        </div>
+    `;
+}
+
+const renderUser = (user) => {
```

Comment:
Maybe change it to `renderUserProfile` or something like it. Confusing, looks like the previous function, but slightly different.